backend/backend/services/openAI/graph_prompt_factory.py

```diff
@@ -0,0 +1,40 @@
+
+import importlib
+from backend.services.openAI.prompts import PROTOCOL_TO_PROMPTS
+PRE_PROMPT = """
+You are an AI that helps write GraphQL queries on the Graph Protocol.
+In the coming prompts I'll feed you questions that you need to turn into graphQL queries that work.
+Note that it's important that if you don't have some specific data (like dates or IDs), just add placeholders.
+Show only code and do not use sentences.
+{}"""
+
+SAMPLE_QUERIES_PRE_PROMPT = """
+Here are a list of example queries based on various graphQL schemas.
+{}"""
+
+
+
+    # subgraph = "aave-governance"
+
+
+class GraphPromptFactory:
+    def __init__(self, protocol):
+        self.protocol = protocol
```

Comment:
Let's pass in `self.chain` here as well. We might want to query by chain i.e. Polygon for the NFT analytics bounty.